src/Nethermind/Nethermind.Blockchain/Contracts/Contract.ConstantContract.cs

```diff
@@ -85,7 +85,7 @@ public override object[] Call(CallInfo callInfo)
 
                 lock (_readOnlyTxProcessorSource)
                 {
-                    using var scope = _readOnlyTxProcessorSource.Build(GetState(callInfo.ParentHeader));
+                    using var scope = _readOnlyTxProcessorSource.BuildAndInit(GetState(callInfo.ParentHeader));
```

Comment:
here we create a new scope, that calls WorldState.BeginScope()

then nested inside the call there is another SystemCall to read the contract address for this call, there we create another scope and that is what create the issues of calling BeginScope twice on the same WorldState
@asdacap